sdk/core/azure-core-test/src/main/java/com/azure/core/test/implementation/RestProxyTests.java

```diff
@@ -1619,20 +1634,125 @@ public void service21GetBytes100() {
     @Host("http://localhost")
     @ServiceInterface(name = "DownloadService")
     interface DownloadService {
+
         @Get("/bytes/30720")
-        StreamResponse getBytes();
+        StreamResponse getBytes(Context context);
+
+        @Get("/bytes/30720")
+        Mono<StreamResponse> getBytesAsync(Context context);
 
         @Get("/bytes/30720")
         Flux<ByteBuffer> getBytesFlux();
     }
 
-    @Test
-    public void simpleDownloadTest() {
-        StepVerifier.create(Flux.using(() -> createService(DownloadService.class).getBytes(),
+    @ParameterizedTest
+    @MethodSource("downloadTestArgumentProvider")
+    public void simpleDownloadTest(Context context) {
+        StepVerifier.create(Flux.using(() -> createService(DownloadService.class).getBytes(context),
                 response -> response.getValue().map(ByteBuffer::remaining).reduce(0, Integer::sum),
                 StreamResponse::close))
             .assertNext(count -> assertEquals(30720, count))
             .verifyComplete();
+
+        StepVerifier.create(Flux.using(() -> createService(DownloadService.class).getBytes(context),
```

Comment:
nit: why not assert both is the same request?

Reply:
I did attempt that, but figured that the test is a bit easier to understand if they're separated, i.e. the reduction logic is simpler this way.